src/pages/billing/Project.js

```diff
@@ -318,36 +310,28 @@ const ProjectDetail = ({ project, billingAccounts, authorizeAndLoadAccounts }) =
     datasetName: selectedDatasetName
   }))
 
-  const refresh = _.flow(
+  const updateBillingProjectUsers = _.flow(
     withErrorReporting('Error loading billing project users list'),
     Utils.withBusyState(setRefreshing)
-  )(async () => {
-    setAddingUser(false)
-    setDeletingUser(false)
-    setUpdating(false)
-    setEditingUser(false)
-    const rawProjectUsers = await Ajax(signal).Billing.project(billingProject.projectName).listUsers()
-    const projectUsers = _.flow(
+  )(() => Ajax(signal).Billing.project(billingProject.projectName).listUsers()
+    .then(_.flow(
       _.groupBy('email'),
       _.map(gs => ({ ..._.omit('role', gs[0]), roles: _.map('role', gs) })),
       _.sortBy('email')
-    )(rawProjectUsers)
-    setProjectUsers(projectUsers)
-  })
+    ))
+    .then(setProjectUsers))
```

Comment:
Could you convert this to async/await? Also, I think setProjectUsers could go in the flow? You could pass the awaited ajax result directly to the flow if you want to avoid sticking it somewhere with a name.
Like:
```js
async () => {
  _.flow(
    // group, map, sort, etc
    setProjectUsers
  )(await Ajax...)
}

Reply:
it can't - you'll get a linter error about an unused return

Reply:
I'm hesitant to convert to async/await in these cases. I think that 
```javascript
return getMeAFoo()
    .then(makeMeABar)
    .the(makeMeABaz)
```
is less busy and consequently reads better than
```javascript
const foo = await getMeAFoo()
const bar = await makeMeABar(foo)
return await makeMeABaz(bar)
```
I also think that the names given to return values in such cases offer very little in terms of clarity, only breaking up the line.

I'm not saying I won't use async/await - it certainly has its place and can make code look much clearer, especially when interleaving control logic and requests. In the simple cases, however, I don't agree that it helps.